docker/Dockerfile.omp_offload

```diff
@@ -0,0 +1,110 @@
+##############################################################
+# This Dockerfile contains the additional NVIDIA compilers, 
+# libraries, and plugins to enable OpenMP offloading and NVIDIA GPU 
+# acceleration of Devito codes.
+#
+# TOFIX: Uncomment line "self.cflags += ['-Xopenmp-target', '-march=sm_37']" in devito/arch/compiler.py and change sm_37 for sm_XX where XX is the compute capability of your GPU
+# BUILD: 
+#   sudo docker build -t devito/ompoffload:1.0 -f docker/Dockerfile.omp_offload .
+# 
+# RUN
+#   sudo docker run --gpus all --rm -it devito/ompoffload:1.0
+# python examples/seismic/acoustic/acoustic_example.py
+
+FROM nvidia/cuda:10.1-devel-ubuntu18.04
+
+ENV DEBIAN_FRONTEND noninteractive
+CMD ["/bin/bash"]
+
+## INSTALL PREREQUISITES
+
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends apt-utils
+RUN apt-get install -y --no-install-recommends build-essential
+RUN apt-get install -y --no-install-recommends libelf-dev
+RUN apt-get install -y --no-install-recommends libffi-dev
+RUN apt-get install -y --no-install-recommends pkg-config
+RUN apt-get install -y --no-install-recommends libnuma-dev
+RUN apt-get install -y --no-install-recommends gcc-multilib
+RUN apt-get install -y --no-install-recommends g++-multilib
+RUN apt-get install -y --no-install-recommends wget
+# latest cmake
+RUN apt-get install -y --no-install-recommends software-properties-common
+RUN apt-get install -y --no-install-recommends lsb-release
+RUN wget -q -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
+RUN apt-add-repository "deb https://apt.kitware.com/ubuntu/ $(lsb_release -cs) main"
+RUN apt-get update
+RUN apt-get -y --no-install-recommends install cmake
+
+## INSTALL CLANG 13
+
+RUN wget -q https://github.com/llvm/llvm-project/archive/refs/tags/llvmorg-13.0.0.tar.gz
+RUN tar -xf llvmorg-13.0.0.tar.gz
+WORKDIR /llvm-project-llvmorg-13.0.0
+RUN mkdir build
+WORKDIR /llvm-project-llvmorg-13.0.0/build
+RUN cmake ../llvm/ -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES=all -DLLVM_ENABLE_PROJECTS="clang" -DLLVM_ENABLE_RUNTIMES="openmp" -DCMAKE_INSTALL_PREFIX=/llvm13
+RUN make -j $(( $(lscpu | awk '/^Socket\(s\)/{ print $2 }') * $(lscpu | awk '/^Core\(s\) per socket/{ print $4 }') ))
+RUN make install
+WORKDIR /
+RUN rm -rf llvmorg-13.0.0.tar.gz llvm-project-llvmorg-13.0.0
+ENV PATH /llvm13/bin:${PATH}
+ENV LD_LIBRARY_PATH /llvm13/lib:${LD_LIBRARY_PATH}
+
+## INSTALL PYTHON
+
+RUN apt-get install -y --no-install-recommends python3-dev
+RUN apt-get install -y --no-install-recommends python3-venv
+
+## INSTALL DEVITO
+
+ADD ./requirements.txt /app/requirements.txt
+ADD ./requirements-optional.txt /app/requirements-optional.txt
+# ADD ./requirements-mpi.txt /app/requirements-mpi.txt
+
+RUN python3 -m venv /venv
```

Comment:
We don't usually use `venv`. And since the docker image is a standalone setup I'm not sure it adds much so would stick to standard python.